src/app/exams/exams-add.component.ts

```diff
@@ -216,7 +216,8 @@ export class ExamsAddComponent implements OnInit {
       this.showErrorMessage();
       return;
     }
-    this.dialog.open(ExamsPreviewComponent, { data: { exam: this.examForm.value }, minWidth: '75vw' });
+    this.examType = params;
```

Comment:
Rather than pass a parameter to the function and modify the component property, simply pass the component property to the dialog.

This confuses things.  The only time `showPreviewDialog` is called `this.examType` is the parameter so this assignment is redundant.

Reply:
Thanks for that! I hadn't looked at it in that way, I believe it is ok now